pyprocessing/math.py

```diff
@@ -1,31 +1,39 @@
 import math
 import random
-
-from pyprocessing.calculation import lerp
-
+import re
 
 __all__ = ('PVector', )
+SWIZZLE_RE = re.compile('^[xyzwrgba]+$')
+SWIZZLE_ORDER = {'x': 0,
+                 'y': 1,
+                 'z': 2,
+                 'w': 3,
+                 'r': 0,
+                 'g': 1,
+                 'b': 2,
+                 'a': 3}
 
 
 class PVectorMeta(type):
     def __getattr__(self, attr):
         if attr == 'add':
             return self._class_add
-        if attr == 'dist':
+        elif attr == 'dist':
```

Comment:
elifs are unnecessary.

Reply:
elifs are a bit faster since they stop at the first match.

Reply:
elifs are slower than ifs. I just benchmarked it to make sure.

```py
>>> import time
>>> def timeit(f):
...     start = time.time_ns()
...     for i in range(1000):
...             f()
...     return time.time_ns()  - start
...
>>> def elifs():
...     if var == 'a':
...             return 'a'
...     elif var == 'b':
...             return 'b'
...     elif var == 'c':
...             return 'c'
...     elif var == 'd':
...             return 'd'
...     elif var == 'e':
...             return 'e'
...     else:
...             return 'f'
...
>>> var = 'e'
>>> def ifs():
...     if var == 'a':
...             return 'a'
...     if var == 'b':
...             return b
...     if var == 'c':
...             return 'c'
...     if var == 'd':
...             return 'd'
...     if var == 'e':
...             return 'e'
...     return 'f'
...
>>> timeit(elifs)
0
>>> timeit(ifs)
501500
>>> timeit(elifs)
0
>>> timeit(elifs)
501700
>>> timeit(elifs)
501800
>>> timeit(elifs)
497900
>>> timeit(elifs)
501800
>>> timeit(elifs)
0
>>> timeit(elifs)
506700
>>> def avg(l):
...     return sum(l) / len(l)
...
>>> rel = [timeit(elifs) for i in range(100)]
>>> rel
[0, 1002900, 0, 0, 0, 1001800, 0, 0, 1003200, 0, 0, 0, 1031300, 0, 0, 1001200, 0, 0, 0, 1002900, 0, 0, 0, 1101500, 0, 529800, 0, 502000, 0, 473400, 0, 0, 0, 1003900, 0, 0, 1002600, 0, 0, 0, 1002700, 0, 0, 0, 1002600, 0, 0, 0, 1030400, 0, 0, 0, 1002900, 0, 0, 0, 1002700, 0, 0, 0, 1002600, 0, 0, 0, 1060800, 0, 502400, 0, 501200, 0, 501100, 0, 0, 1004200, 0, 0, 0, 1029800, 0, 0, 1005000, 0, 0, 0, 1000600, 0, 0, 0, 1002600, 0, 0, 0, 1002100, 0, 0, 0, 1003200, 0, 0, 0]
>>> rel = [x for x in rel if x]
>>> rel
[1002900, 1001800, 1003200, 1031300, 1001200, 1002900, 1101500, 529800, 502000, 473400, 1003900, 1002600, 1002700, 1002600, 1030400, 1002900, 1002700, 1002600, 1060800, 502400, 501200, 501100, 1004200, 1029800, 1005000, 1000600, 1002600, 1002100, 1003200]
>>> avg(rel)
907358.6206896552
>>> rif = [timeit(ifs) for i in range(100)]
>>> rif = [x for x in rif if x]
>>> avg(rif)
501788.8888888889
>>> avg(rel) < avg(rif)
False
>>> avg(rel) - avg(rif)
405569.7318007663
```

Reply:
well, in that case.

Reply:
just so you know, in this case, ifs are faster because the parser doesn't have to compute every case to find the first matching one (it only works because the function returns). In that test, there are 6 'blocks' of code for the parser to interpret, while in the elif case, there's only one big block. The difference is marginal (406 ms accross 1,000 calls) but it's still a bit faster, plus it looks a bit better in my opinion

Reply:
Thats interesting, never thought about that.